docs/datastore/datastore-api.md

```diff
@@ -0,0 +1,32 @@
+---
+id: datastore-api
+title: DataStore
+sidebar_label: DataStore
+---
+
+The `DataStore` object initializes the storage layer, replication layer and the Models.
+It accepts some options on creation.
+
+| Input | Type | Description |
+| ----- | ---- | ----------- |
+| config | [DataStoreConfig](#DataStoreConfig) | Configuration Options |
+| customEngines | ?[CustomEngies](#CustomEngines) | Custom storage and replication implementation |
+
+### DataStoreConfig
+
+Configuration options for `DataStore`.
+
+| Input | Type | Description |
+| ----- | ---- | ----------- |
+| dbName | ?string | The Database name |
+| schemaVersion | ?number | The Schema Version number. Used to trigger a Schema upgrade. Defaults to 1 |
+| replicationConfig | ?[GlobalReplicationConfig](replication-api#GlobalReplicationConfig) | Configuration for Replication engine |
+
+### CustomEngines
```

Comment:
This is internal - it will be covered in the platform specific config (react native etc.)